buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -1006,6 +1006,13 @@ private AbstractByteBuf rootParent() {
             throw new IllegalReferenceCountException();
         }
 
+        private ByteBuffer getTmpNioBuf() {
+            if (tmpNioBuf == null) {
+                tmpNioBuf = rootParent().internalNioBuffer(adjustment, length).slice();
+            }
+            return tmpNioBuf;
+        }
```

Comment:
add a `cleanTmpNioBuf()` method

Reply:
This method got inlined into the only place it's called. I don't think there's a need for a separate clear method.